app/controllers/concerns/test_track/controller.rb

```diff
@@ -15,6 +15,12 @@ def require_feature_flag(feature_flag, *args)
         raise ActionController::RoutingError, 'Not Found' unless test_track_visitor.ab(feature_flag, context: self.class.name.underscore)
       end
     end
+
+    def prohibit_feature_flag(feature_flag, *args)
+      before_action(*args) do
+        raise ActionController::RoutingError, 'Not Found' if test_track_visitor.ab(feature_flag, context: self.class.name.underscore)
+      end
+    end
```

Comment:
Have y'all considered that this could be the same method, but we could add an optional `variant:` kwarg to it, to relax the assumption we've made that they always want to match on the true variant? If this is functionality we want, maybe we want users to be explicit about which variant they want, and abandon that assumption. This would also allow for us to more easily expand to support non-traditional cases of `true_variant` not being `true` (which isn't currently supported here, from what I can tell).

```ruby
def require_feature_flag(feature_flag, *args, variant: true)
  before_action(*args) do
    raise ActionController::RoutingError, 'Not Found' unless test_track_visitor.ab(feature_flag, context: self.class.name.underscore) == variant
  end
end
```

Maybe even better to not give it a default, and introduce it as a breaking change that users are required to include a variant, but that may also just be a more straightforward interface for developers

Reply:
Having a rock-solid and opinionated experience for true/false feature gating feels preferable to expanding the required interface for the rare cases when we want to gate on a variant other than "true." I'm not opposed to adding some extensibility here (especially in cases where we might want to gate an entire controller under a branch of an experiment) but my vote would be to default it to "true" for any true feature gates (i.e. if the name ends in `_enabled`).

Reply:
I didn't consider that, but I do like the idea! I think I immediately went to a new method because it was so ingrained in my head that `require_feature_flag` meant it **had** to be `true`. I'm definitely open to expanding on `require_feature_flag` if that's what the majority deems preferable!